acra-censor/handlers/handlers_util.go

```diff
@@ -2,37 +2,34 @@ package handlers
 
 import (
 	"errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/xwb1989/sqlparser"
+	"github.com/xwb1989/sqlparser/dependency/querypb"
 	"strings"
 )
 
 var ErrQueryNotInWhitelist = errors.New("query not in whitelist")
 var ErrQueryInBlacklist = errors.New("query in blacklist")
-
 var ErrAccessToForbiddenTableBlacklist = errors.New("query tries to access forbidden table")
 var ErrAccessToForbiddenTableWhitelist = errors.New("query tries to access forbidden table")
-
 var ErrForbiddenSqlStructureBlacklist = errors.New("query's structure is forbidden")
 var ErrForbiddenSqlStructureWhitelist = errors.New("query's structure is forbidden")
-
 var ErrParseSqlRuleBlacklist = errors.New("parsing security rules error")
-
 var ErrParseSqlRuleWhitelist = errors.New("parsing security rules error")
-
 var ErrNotImplemented = errors.New("not implemented yet")
-
 var ErrQuerySyntaxError = errors.New("fail to parse specified query")
-
 var ErrComplexSerializationError = errors.New("can't perform complex serialization of queries")
 var ErrSingleQueryCaptureError = errors.New("can't capture single query")
 var ErrCantOpenFileError = errors.New("can't open file to write queries")
 var ErrCantReadQueriesFromFileError = errors.New("can't read queries from file")
 var ErrUnexpectedCaptureChannelClose = errors.New("unexpected channel closing while query logging")
-
 var ErrUnexpectedTypeError = errors.New("should never appear")
 
-const (
-	LogQueryLength = 100
-)
+const LogQueryLength = 100
+const ValuePlaceholder = "replaced"
+
+//global logging object for censor
+var Logger = log.WithFields(log.Fields{"service": "acra-censor"})
```

Comment:
I suggest to avoid using singletons and global variables. In such keys any other code that use this logger can change his state:
```
func MyNewHandler()error{
  acracensor.Logger = acracensor.Logger.WithField("handler", "MyHandler")
  ...
  if err != nil {
     acracensor.Logger.Errorln("blabla")
     return err
  }
}
```
at first, we can encapsulate initialization of logger and his creation, as example to return copy
```
// logger is private variable
var logger = log.WithFields(log.Fields{"service": "acra-censor"})
func GetLogger() *logrus.Entry{
  // return copy of logger
}
```
```
func MyHandler(){
   logger = acracensor.GetLogger().WithField("handler", "MyHandler")
   acracensor.GetLogger().Errorln("bla bla)
}
```
Better if all pieaces will log in their own scope. We can use dependency injection:
```
type MyHandler struct{
... 
logger *logrus.Entry
...
}
func MyNewHandler(blabla, logger *logrus.Entry)(*MyHandler, error){
  return &MyHandler{logger: logger.WithField("handler", "MyHandler")}, nil
}
```

and then use it internally to this handler
```
func (handler *MyHandler) CheckQuery(query string) error {
 ...
  handler.logger.Errorln("WTF?")
}
```

for the first time I suggest to create own logger for each handler and censor and use them internally